changelog/unreleased/issue-19693.toml

```diff
@@ -0,0 +1,4 @@
+type="c"
+message="Changed default setting for Kafka transports to non-legacy mode."
```

Comment:
Please add a note to `UPGRADING.md` for this change. Thanks!

Reply:
Done!